### ISSUE TITLE
MultiLineEntry improperly displayed when having long text inside Scroll container

### ISSUE THREAD

**Issue by wsdexy** [fyne-io/fyne#2814]
### Describe the bug:
MultiLineEntry improperly displayed when having long text inside Scroll container, scroll down and you will see it.

### To Reproduce:
Run the Example code.

### Screenshots:
![bug_screen1](https://user-images.githubusercontent.com/17593203/156194507-56c39c6b-df6b-48a4-b2a4-e029d9e11ced.PNG)
![bug_screen2](https://user-images.githubusercontent.com/17593203/156194517-50595ce7-e50a-4306-a988-17df57a31a09.PNG)

### Example code:
```go
appInstance := app.NewWithID("appInstance")
win := appInstance.NewWindow("window")

longText := ""
for i := 0; i < 1024; i++ {
    longText += fmt.Sprintf("%v ", i)
}

entry := widget.NewMultiLineEntry()
entry.Wrapping = fyne.TextWrapWord
//entry.Disable()
entry.SetText(longText)

formItemSlice := []*widget.FormItem{
    &widget.FormItem{
        Text:     "entry",
        Widget:   entry,
        HintText: "",
    },
}
for i := 0; i < 32; i++ {
    formItemSlice = append(formItemSlice, &widget.FormItem{
        Text:     "Label",
        Widget:   widget.NewLabel("Label"),
        HintText: "Label",
    })
}

form := widget.NewForm(
    formItemSlice...,
)

win.SetContent(container.NewVScroll(form))
win.Resize(fyne.NewSize(800, 600))
win.ShowAndRun()
```

### Device (please complete the following information):
 - **OS:** win10
 - **Version:** 21H1
 - **Go version:** 1.17
 - **Fyne version:** v2.0

**Comment by andydotxyz** [comment]
Can you please try updating to Fyne v2.1.3? The text, Entry and Scrolling have been updated since last January.

**Comment by wsdexy** [comment]
![image](https://user-images.githubusercontent.com/17593203/156715863-eb887982-ee47-4fcb-aae4-a0cf1ccd2b63.png)
@andydotxyz 
Thanks for replay, updated to Fyne v2.1.3, still improperly displayed.

**Comment by ranjith-bhide** [comment]
Hi All, 
I was able to find the root cause of this issue. 
I find this happening when we have a scroll container inside a scroll container. 
Whenever the Parent Scroll Container is moved such that the child container is out of the visible range of parent, the function `StartClipping()` from painter.go is called for the child with a negative width/height Value.
this in turn makes an OpenGL binding call to `void glScissor()` function. Negative width/height value is an Error condition for this function as per the [reference manual](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glScissor.xhtml) 
> Errors
GL_INVALID_VALUE is generated if either width or height is negative.

Thus it seems to be causing this undesired behaviour.
Here is my proposed Fix:
```diff
diff --git a/internal/painter/gl/gl_core.go b/internal/painter/gl/gl_core.go
index de68f739..274df863 100644
--- a/internal/painter/gl/gl_core.go
+++ b/internal/painter/gl/gl_core.go
@@ -204,6 +204,8 @@ func (p *glPainter) glClearBuffer() {
 }
 
 func (p *glPainter) glScissorOpen(x, y, w, h int32) {
+       w = int32(math.Max(float64(w), 0))
+       h = int32(math.Max(float64(h), 0))
        gl.Scissor(x, y, w, h)
        gl.Enable(gl.SCISSOR_TEST)
        logError()
```

I have verified that the Issue is no longer reproducible with this change.

PS: I was not able to think of a more cleaner way to clamp the w and h variable to be non-negative. Any suggestions are welcome :)

**Comment by andydotxyz** [comment]
It looks to me that we should skip rendering it if we know the scissor will be 0 size (or less).
Good catch though!

**Comment by ranjith-bhide** [comment]
> It looks to me that we should skip rendering it if we know the scissor will be 0 size (or less).

Yes, That is what is expectated.
and this would be one way of getting OpenGL Scissors to do it for us, following the generic design, and it may also handle some other cases if the Scissors API is used elsewhere with negative width/height.

If you feel its better to skip the rendering instead, Please point me towards the right place file/function where I can add this kind of a check. Will Update the PR with appropriate changes

[Copy-Pasted Comment from the PR]

**Comment by andydotxyz** [comment]
Fixed on `develop` ready for v2.2.0, thanks @ranjith-bhide
